src/uu/pathchk/src/pathchk.rs

```diff
@@ -12,117 +12,99 @@
-    opts.optflag("V", "version", "output version information and exit");
-    let matches = match opts.parse(&args[1..]) {
-        Ok(m) => m,
-        Err(e) => crash!(1, "{}", e),
-    };
+pub fn uumain(args: impl uucore::Args) -> i32 {
+    let usage = get_usage();
+
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(&usage[..])
+        .arg(
+            Arg::with_name(options::POSIX)
+                .short("p")
+                .help("check for most POSIX systems"),
+        )
+        .arg(
+            Arg::with_name(options::POSIX_SPECIAL)
+                .short("P")
+                .help(r#"check for empty names and leading "-""#),
+        )
+        .arg(
+            Arg::with_name(options::PORTABILITY)
+                .long(options::PORTABILITY)
+                .help("check for all POSIX systems (equivalent to -p -P)"),
+        )
+        .arg(Arg::with_name(options::PATH).hidden(true).multiple(true))
+        .get_matches_from(args);
 
     // set working mode
-    let mode = if matches.opt_present("version") {
-        Mode::Version
-    } else if matches.opt_present("help") {
-        Mode::Help
-    } else if (matches.opt_present("posix") && matches.opt_present("posix-special"))
-        || matches.opt_present("portability")
-    {
+    let is_posix = matches.values_of(options::POSIX).is_some();
+    let is_posix_special = matches.values_of(options::POSIX).is_some();
+    let is_portability = matches.values_of(options::PORTABILITY).is_some();
+
+    let mode = if (is_posix && is_posix_special) || is_portability {
         Mode::Both
-    } else if matches.opt_present("posix") {
+    } else if is_posix {
         Mode::Basic
-    } else if matches.opt_present("posix-special") {
+    } else if is_posix_special {
         Mode::Extra
     } else {
         Mode::Default
     };
 
     // take necessary actions
-    match mode {
-        Mode::Help => {
-            help(opts);
-            0
-        }
-        Mode::Version => {
-            version();
-            0
-        }
-        _ => {
-            let mut res = if matches.free.is_empty() {
-                show_error!("missing operand\nTry {} --help for more information", NAME);
-                false
-            } else {
-                true
-            };
-            // free strings are path operands
-            // FIXME: TCS, seems inefficient and overly verbose (?)
-            for p in matches.free {
-                let mut path = Vec::new();
-                for path_segment in p.split('/') {
-                    path.push(path_segment.to_string());
-                }
-                res &= check_path(&mode, &path);
-            }
-            // determine error code
-            if res {
-                0
-            } else {
-                1
+    let paths = matches.values_of(options::PATH);
+    let mut res = if paths.is_none() {
```

Comment:
I think this can be done with min_values in clap, no?

Reply:
`min_values` is to set min occurrences of argument values. It doesn't check whether the argument itself is specified at least once.

from https://docs.rs/clap/2.32.0/clap/struct.Arg.html#method.min_values
> ...
> NOTE: This does not implicitly set Arg::multiple(true). This is because -o val -o val is multiple occurrences but a single value and -o val1 val2 is a single occurrence with multiple values. ...

Calling `required(true)` on the argument makes sure it is specified at least once. But, when the argument is not specified it is outputting the following error
```
error: The following required arguments were not provided:
    <path>...

USAGE:
    pathchk [OPTION]... NAME...

For more information try --help
```
The error message is different from the existing message. Moreover, the mention of `<path>` doesn't match with the usage help.

So, I thought of doing the check explicitly.